src/components/EnvoyStats/EnvoyStats.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <StatusInfo
+    :has-error="hasError"
+    :is-loading="isLoading"
+  >
+    <KCard border-variant="noBorder">
+      <template v-slot:body>
+        <Prism
+          id="stats"
+          language="json"
+          :code="stats"
+        />
+      </template>
+      <template v-slot:actions>
+        <KClipboardProvider
+          v-if="stats"
+          v-slot="{ copyToClipboard }"
+        >
+          <KPop placement="bottom">
+            <KButton
+              appearance="primary"
+              @click="() => { copyToClipboard(stats) }"
+            >
+              Copy config to clipboard
+            </KButton>
+            <template v-slot:content>
+              <div>
+                <p>Config copied to clipboard!</p>
+              </div>
+            </template>
+          </KPop>
+        </KClipboardProvider>
+      </template>
+    </kcard>
+  </StatusInfo>
+</template>
+
+<script>
+import Kuma from '@/services/kuma'
+import StatusInfo from '@/components/Utils/StatusInfo'
+import Prism from 'vue-prismjs'
+
+export default {
+  name: 'EnvoyStats',
+  components: {
+    StatusInfo,
+    Prism,
+  },
+  props: {
+    mesh: {
+      type: String,
+      default: '',
+    },
+    dppName: {
+      type: String,
+      default: '',
+    },
+    zoneIngressName: {
+      type: String,
+      default: '',
+    },
+    zoneEgressName: {
+      type: String,
+      default: '',
+    },
+  },
+
+  data() {
+    return {
+      isLoading: true,
+      hasError: false,
+    }
+  },
+
+  watch: {
+    dppName() {
+      this.fetchStats()
+    },
+    zoneIngressName() {
+      this.fetchStats()
+    },
+    zoneEgressName() {
+      this.fetchStats()
+    },
+  },
+  mounted() {
+    this.fetchStats()
+  },
+
+  methods: {
+    async fetchStats() {
+      this.hasError = false
+      this.isLoading = true
+
+      try {
+        let stats = ''
+
+        if (this.mesh !== '' && this.dppName !== '') {
```

Comment:
This `else if`-ing on the type of stats we want to fetch looks a bit weird to me. Why don't we know which stats we want to download / how do we make sure that these properties are empty on zoneIngress / zoneEgress stats?

Reply:
we don't. You can define mesh, dppname, zoneingress and zoneegress and we will fetch then the dppName.

I personally don't see this as a problem since we are the only users of this component, but if someone has suggestions on how to implement this without adding tens of lines of code with validation then I'm happy to change it.